### ISSUE TITLE
Failed: DDataClusterShardingLeaving: must recover after leaving coordinator node

### ISSUE THREAD

**Issue by octonato** [akka/akka#28743]
Probably same as #22632. Slowness? 

https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/2291/console

```scala
info] [JVM-1] - must recover after leaving coordinator node (on node 'first', class akka.cluster.sharding.DDataClusterShardingLeavingMultiJvmNode1) *** FAILED ***
[info] [JVM-1]   java.lang.AssertionError: assertion failed: timeout (15 seconds) during expectMsg: Terminated Actor[akka://MultiNodeClusterShardingSpec/system/sharding/Entity#1313116709]
[info] [JVM-1]   at scala.Predef$.assert(Predef.scala:223)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectMsgPF(TestKit.scala:437)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectMsgPF$(TestKit.scala:434)
[info] [JVM-1]   at akka.testkit.TestKit.expectMsgPF(TestKit.scala:929)
```

followed by: 

```scala
[info] [JVM-3] - must recover after leaving coordinator node (on node 'third', class akka.cluster.sharding.DDataClusterShardingLeavingMultiJvmNode3) *** FAILED ***
[info] [JVM-3]   Actor[akka://MultiNodeClusterShardingSpec@a4.moxie:49650/system/sharding/Entity/8/8#-1804455747] was equal to Actor[akka://MultiNodeClusterShardingSpec@a4.moxie:49650/system/sharding/Entity/8/8#-1804455747] (ClusterShardingLeavingSpec.scala:135)
[info] [JVM-2] - must recover after leaving coordinator node (on node 'second', class akka.cluster.sharding.DDataClusterShardingLeavingMultiJvmNode2) *** FAILED ***
[info] [JVM-2]   Actor[akka://MultiNodeClusterShardingSpec@a4.moxie:49650/system/sharding/Entity/8/8#-1804455747] was equal to Actor[akka://MultiNodeClusterShardingSpec@a4.moxie:49650/system/sharding/Entity/8/8#-1804455747] (ClusterShardingLeavingSpec.scala:135)
[info] [JVM-4] - must recover after leaving coordinator node (on node 'fourth', class akka.cluster.sharding.DDataClusterShardingLeavingMultiJvmNode4) *** FAILED ***
[info] [JVM-4]   Actor[akka://MultiNodeClusterShardingSpec@a4.moxie:49650/system/sharding/Entity/8/8#-1804455747] was equal to Actor[akka://MultiNodeClusterShardingSpec@a4.moxie:49650/system/sharding/Entity/8/8#-1804455747] (ClusterShardingLeavingSpec.scala:135)
```

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/2304/consoleFull

**Comment by raboof** [comment]
https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/2326/consoleFull

**Comment by renatocaval** [comment]
again: https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/2328/consoleFull

**Comment by seglo** [comment]
https://jenkins.akka.io:8498/job/akka-classic-cluster-tests/378/consoleFull

**Comment by raboof** [comment]
Similar failure at https://jenkins.akka.io:8498/job/akka-multi-node/4110/consoleFull

**Comment by patriknw** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/4116

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7848/consoleText

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7806/consoleText

**Comment by johanandren** [comment]
I wonder if it is failing more often since we merged the new remember entities, or if it's just a coincident

**Comment by seglo** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/4131/consoleFull

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7854/consoleFull

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-classic-cluster-tests/444/

**Comment by patriknw** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-2.13/660/

**Comment by johanandren** [comment]
I think this is the same as #24709 shards are assigned to the leaving nodes so that they never get to complete graceful shutdown of their regions. Should be improved by the cluster awareness in LeastShardAllocationStrategy but it is still weird that a region would start a shard when it is shutting down. I'll see if I can do something about that.

**Comment by johanandren** [comment]
Should be fixed by @octonato and PR #29707
